js/app.js

```diff
@@ -27,19 +27,78 @@
 
 	var App = Marionette.Application.extend({
 		_registerPageEvents: function() {
-			$('#oca-spreedme-add-room').submit(function() {
-				return false;
-			});
+			$('#edit-roomname').select2({
+				ajax: {
+					url: OC.linkToOCS('apps/files_sharing/api/v1') + 'sharees',
+					dataType: 'json',
+					quietMillis: 100,
+					data: function (term) {
+						return {
+							format: 'json',
+							search: term,
+							perPage: 200,
+							itemType: 'folder',
+							shareType: 0
+						};
+					},
+					results: function (response) {
+						// TODO improve error case
+						if (response.ocs.data === undefined) {
+							console.error('Failure happened', response);
+							return;
+						}
+
+						var results = [];
+						$.each(response.ocs.data.users, function(id, user) {
+							results.push({ id: user.value.shareWith});
+						});
 
-			// Create a new room
-			$('#oca-spreedme-add-room > button.icon-confirm').click(function() {
-				var roomname = $('#oca-spreedme-add-room > input[type="text"]').
-					val();
-				if (roomname === "") {
-					return;
+						return {
+							results: results,
+							more: false
+						};
+					}
+				},
+				initSelection: function (element, callback) {
+					callback({id: element.val()});
+				},
+				formatResult: function (element) {
+					return '<span><div class="avatar" data-user="' + escapeHTML(element.id) + '" data-user-display-name="' + escapeHTML(element.id) + '"></div>' + escapeHTML(element.id) + '</span>';
+				},
+				formatSelection: function (element) {
+					return '<span><div class="avatar" data-user="' + escapeHTML(element.id) + '" data-user-display-name="' + escapeHTML(element.id) + '"></div>' + escapeHTML(element.id) + '</span>';
```

Comment:
Btw, why is there a div (block-level element) in a span (inline element) ;) Shouldn’t be like that. Just make both elements divs.